File: a01/Username.java
/*
  prints the current username

  @author: Lothar Rubusch

  compile:
  $ javac ./Username.java

  run:
  $ java Username

  disassemble:
  $ javap -c ./Username.class
*/
public class Username{
    public static void main( String[] args){
        String username = "";
        try{
            username = System.getProperty("user.name");
        }catch(Exception e){
            System.out.println( "FAILED, some exception was thrown" );
            System.exit(0);
        }
        System.out.println( "username '" + username + "'" );
    }
}

File: A01__Username/Username.c
/*
  prints the current username

  @author Lothar Rubusch

  compile:
  $ gcc -m32 ./Username.c -o ./Username.exe

  disassemble:
  $ gcc -m32 ./Username.c -S
 */
#include <stdlib.h>
#include <stdio.h>
int main( int args, char* argv[] ){
	char *username = NULL;
	username = getenv("USER");
	if( NULL != username ){
		printf("username '%s'\n", username );
	}
	exit( EXIT_SUCCESS );
}
